app/assets/javascripts/media_player_wrapper/avalon_player_new.es6

```diff
@@ -284,32 +284,24 @@ class MEJSPlayer {
     // Build quality
     this.player.buildquality(this.player, null, null, this.mediaElement);
 
-    this.reInitializeCaptions(playlistItemsT);
+    this.reInitializeCaptions();
 
-    this.player.load();
+    // this.player.load();
     this.player.play();
   }
 
   /** Based on the availability of captions in currentStreamInfo build tracks
    * in playlists and item page
    * @function reInitializeCaptions
-   * @param playlistItemsT 
    */
-  reInitializeCaptions(playlistItemsT) {
+  reInitializeCaptions() {
     if (this.currentStreamInfo.captions_path) {
-      // Re-arrange player features when tracks are available
-      // in both playlists and item page
-      this.player.featurePosition.tracks = 6;
-      this.player.featurePosition.fullscreen = 9;
+      // Place tracks button after volume button when tracks are available
+      this.player.featurePosition.tracks = this.player.featurePosition.volume + 1;
```

Comment:
@cjcolvar What do think of this solution? I feel this is better than the previous one.

Reply:
:+1: This looks good to me.  Does it automatically push other features one position higher when you set it like this?

Reply:
Yes it does.

Reply:
:+1: Neat!